### ISSUE TITLE
contains() is broken with scatter plots with master, works with v1.4.3rc1

### ISSUE THREAD

**Issue by MathieuDuponchelle** [matplotlib/matplotlib#4076]
```
#!/usr/bin/python3

from gi.repository import Gtk
from matplotlib.figure import Figure
from matplotlib.backends.backend_gtk3cairo import FigureCanvasGTK3Cairo as FigureCanvas
from matplotlib.patches import Rectangle

import signal

class DrawPoints:
    '''Creates random points, 2 axis on 1 figure on 1 canvas on init. Allows for drawing and zooming of points.'''
    def __init__(self):
        self.xs = [0, 500, 1300, 1400, 2000]
        self.ys = [0, 1, 7, 6, 3]

        self.fig = Figure()
        self.ax = self.fig.add_axes ([0, 0, 1, 1], axisbg='None')
        self.canvas = FigureCanvas(self.fig)

    def on_pick (self, event):
        print ("got picked")

    def on_press(self, event):
        print (self.paths.contains (event))

    def draw(self):
        '''Draws the ax-subplot'''
        self.paths = self.ax.scatter (self.xs, self.ys, picker=True)
        self.canvas.mpl_connect ('pick_event', self.on_pick)
        self.cidpress = self.canvas.mpl_connect('button_press_event', self.on_press)

signal.signal (signal.SIGINT, signal.SIG_DFL)

window = Gtk.Window()
window.connect("delete-event", Gtk.main_quit)
window.set_default_size(1000, 600)

box = Gtk.Box(orientation=Gtk.Orientation.VERTICAL)
window.add(box)

scrolled = Gtk.ScrolledWindow ()

points = DrawPoints()
points.draw()

scrolled.add (points.canvas)

box.pack_start(scrolled, True, True, 0)

window.show_all()
Gtk.main()
```

**Comment by MathieuDuponchelle** [comment]
I tried to bisect but ended up on a bunch of commits where build was failing, and too wide a range of commits to hope reverting them all.

**Comment by tacaswell** [comment]
What do you mean by 'broken' and can you simplify this example at all?

**Comment by MathieuDuponchelle** [comment]
By broken I mean that when receiving the click events after carefully clicking on one of the markers, self.paths.contains(event) returns (False, empty_array) whereas it does return True and an array correctly populated with 1.4.3rc1.

As for the example, I think it is simple enough :)

**Comment by tacaswell** [comment]
Yes, but I don't have GTK set up in my normal dev environment.  By simplify I mean 'make gui framework independent'.

**Comment by MathieuDuponchelle** [comment]
hmph OK, you just need to grab the script there http://matplotlib.org/examples/event_handling/pick_event_demo.html and click on the dots in figure 3, the problem is essentially the same (I think I'm safe in assuming pick_event is emitted after calls to contains() return True)

**Comment by efiring** [comment]
Confirmed with Figure 3 from the pick_event_demo.

**Comment by efiring** [comment]
I have not been able to bisect this all the way on my Ubuntu VM; I ended up with 50 commits that I had to skip.  This range includes the removal of CXX.  Using good old print statements, however, I see that in Collection.contains(), everything is looking plausible until near the end: the call to _path.point_in_path_collection() is returning an empty list.  Either its arguments are now incorrect, or that function is broken.  I suspect the latter, because it doesn't look like anything would have changed the arguments. @mdboom?
